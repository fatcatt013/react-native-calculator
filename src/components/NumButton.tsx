import React, { FC } from 'react';
import { View, Text, StyleSheet, TouchableHighlight } from 'react-native';

const NumButton = (props: {
  num: string;
  handleFunc(btn: string): void;
  last: boolean;
  special?: boolean;
}) => {
  return (
    <View style={props.special ? styles.specialContainer : styles.container}>
      <TouchableHighlight
        style={styles.touchable}
        underlayColor={props.last ? '#0CABA8' : '#589A8D'}
        onPress={() => {
          props.handleFunc(props.num);
        }}
        testID={props.num}
      >
        <Text style={{ fontSize: 30, color: 'white' }}>{props.num}</Text>
      </TouchableHighlight>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  specialContainer: {
    flex: 0.3,
  },
  touchable: {
    flex: 1,
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default NumButton;
