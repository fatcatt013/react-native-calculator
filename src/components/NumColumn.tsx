import React, { FC } from 'react';
import {
  NativeModules,
  View,
  StyleSheet,
  TouchableHighlight,
} from 'react-native';
import NumButton from './NumButton';

const NumColumn = (props: {
  nums: string[];
  handleFunc(btn: string): void;
}) => {
  return (
    <View style={styles.container}>
      {props.nums.map((num, key) => {
        return <NumButton handleFunc={props.handleFunc} num={num} key={key} />;
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '25%',
  },
});

export default NumColumn;
