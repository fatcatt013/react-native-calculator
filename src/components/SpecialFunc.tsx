import React, { FC } from 'react';
import { View, Text, StyleSheet, ProgressViewIOSComponent } from 'react-native';
import NumButton from './NumButton';

const SpecialFunc = (props: { handleFunc(btn: string): void }) => {
  return (
    <View style={styles.container}>
      <NumButton
        num={'DEL'}
        handleFunc={props.handleFunc}
        last={true}
        special={true}
      />
      <NumButton
        num={'CE'}
        handleFunc={props.handleFunc}
        last={true}
        special={true}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#008F8C',
    flex: 0.06,
    flexDirection: 'row',
    justifyContent: 'space-between',
    width: '100%',
  },
});

export default SpecialFunc;
