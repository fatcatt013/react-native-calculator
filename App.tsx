import React, { FC, useState } from 'react';
import {
  StyleSheet,
  Text,
  useWindowDimensions,
  View,
  StatusBar,
} from 'react-native';
import NumColumn from './src/components/NumColumn';
import SpecialFunc from './src/components/SpecialFunc';

export default function App() {
  const [currNum, setCurrNum] = useState<string>('');
  const [calculation, setCalculation] = useState<string[]>([]);
  const [calculated, setCalculated] = useState<boolean>(false);
  const [calculateNextRender, setCalculateNextRender] = useState(false);

  const calculate = (): void => {
    setCalculation([...calculation, currNum]);

    let temp: number = parseInt(calculation[0]);
    for (let i = 0; i < calculation.length - 2; ) {
      const firstNum: number = temp;
      const operator = calculation[i + 1];
      const secondNum: number = parseInt(calculation[i + 2]);

      switch (operator) {
        case '+':
          temp = firstNum + secondNum;
          break;
        case '-':
          temp = firstNum - secondNum;
          break;
        case '/':
          temp = firstNum / secondNum;
          break;
        case 'x':
          temp = firstNum * secondNum;
          break;
      }
      i += 2;
    }
    setCurrNum(temp.toString());
    setCalculated(true);
  };

  const handleClick = (btn: string): void => {
    const operators = ['+', '-', 'x', '/'];

    if (calculated) {
      setCalculation([]);
      setCurrNum('');
      setCalculated(false);
      return;
    }

    if (btn === '( - )') {
      if (currNum[0] === '-') {
        setCurrNum(currNum.substring(1, currNum.length));
      } else {
        setCurrNum(`-${currNum}`);
      }
      return;
    }

    if (operators.includes(btn)) {
      if (currNum.length > 0) {
        setCalculation([...calculation, currNum, btn]);
        setCurrNum('');
      }
    } else {
      switch (btn) {
        case 'DEL':
          setCurrNum(currNum.substring(0, currNum.length - 1));
          break;

        case 'CE':
          setCurrNum('');
          setCalculation([]);
          break;

        case '=':
          setCalculation([...calculation, currNum]);
          setCurrNum('');
          setCalculateNextRender(true);
          break;

        default:
          setCurrNum(currNum + btn);
      }
    }
    console.log(calculation);
  };

  if (calculateNextRender) {
    calculate();
    setCalculateNextRender(false);
  }

  return (
    <View style={styles.mainContainer}>
      <View style={styles.header}>
        <Text style={{ fontSize: 20 }}>{calculation.join('')}</Text>
      </View>
      <View style={styles.topContainer}>
        <Text style={{ fontSize: 50 }}>{currNum}</Text>
      </View>
      <SpecialFunc handleFunc={handleClick} />
      <View style={styles.bottomContainer}>
        <NumColumn
          last={false}
          handleFunc={handleClick}
          nums={['7', '4', '1', '.']}
        />
        <NumColumn
          last={false}
          handleFunc={handleClick}
          nums={['8', '5', '2', '0']}
        />
        <NumColumn
          last={false}
          handleFunc={handleClick}
          nums={['9', '6', '3', '=']}
        />
        <NumColumn
          last={true}
          handleFunc={handleClick}
          nums={['( - )', '+', '-', 'x', '/']}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },

  topContainer: {
    backgroundColor: '#8AA6A3',
    flex: 0.3,
    width: '100%',
    paddingTop: StatusBar.currentHeight,
    alignItems: 'flex-end',
    justifyContent: 'flex-end',
    paddingBottom: 10,
    paddingRight: 20,
  },

  bottomContainer: {
    backgroundColor: '#127369',
    flex: 0.6,
    width: '100%',
    flexDirection: 'row',
  },
  header: {
    backgroundColor: '#8AA6A3',
    flex: 0.1,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
