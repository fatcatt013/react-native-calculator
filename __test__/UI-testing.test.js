import TestRenderer from 'react-test-renderer';
import NumButton from '../src/components/NumButton';
import App from '../App';
import SpecialFunc from '../src/components/SpecialFunc';
import NumColumn from '../src/components/NumColumn';

describe('Testing the UI', () => {
  const app = TestRenderer.create(<App></App>).toJSON();

  const normalButton = TestRenderer.create(
    <NumButton
      num={'DEL'}
      last={false}
      special={false}
      handleFunc={(btn) => console.log(btn)}
    />,
  ).toJSON();

  const specialFunc = TestRenderer.create(
    <SpecialFunc handleFunc={(btn) => console.log(btn)} />,
  ).toJSON();

  const numColumn = TestRenderer.create(
    <NumColumn
      nums={['1', '2', '3']}
      handleFunc={(btn) => console.log(btn)}
      last={false}
    />,
  ).toJSON();

  const numColumnLast = TestRenderer.create(
    <NumColumn
      nums={['1', '2', '3']}
      handleFunc={(btn) => console.log(btn)}
      last={true}
    />,
  ).toJSON();

  const lastButton = TestRenderer.create(
    <NumButton
      num={'1'}
      last={true}
      special={false}
      handleFunc={(btn) => console.log(btn)}
    />,
  ).toJSON();

  const specialButton = TestRenderer.create(
    <NumButton
      num={'1'}
      last={true}
      special={false}
      handleFunc={(btn) => console.log(btn)}
    />,
  ).toJSON();

  it('Testing whole App UI', () => {
    expect(app).toMatchSnapshot();
  });

  it('Testing normal button UI', () => {
    expect(normalButton).toMatchSnapshot();
  });

  it('Testing last button UI', () => {
    expect(lastButton).toMatchSnapshot();
  });

  it('Testing special button UI', () => {
    expect(specialButton).toMatchSnapshot();
  });

  it('Testing number column UI', () => {
    expect(numColumn).toMatchSnapshot();
  });

  it('Testing last number column UI', () => {
    expect(numColumnLast).toMatchSnapshot();
  });

  it('Testing special functions UI', () => {
    expect(specialFunc).toMatchSnapshot();
  });
});
