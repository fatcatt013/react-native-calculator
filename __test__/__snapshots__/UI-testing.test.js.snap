// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the UI Testing last button UI 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="1"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`Testing the UI Testing last number column UI 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#008F8C",
      "width": "25%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="1"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="2"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        2
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="3"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        3
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing the UI Testing normal button UI 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="DEL"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
        }
      }
    >
      DEL
    </Text>
  </View>
</View>
`;

exports[`Testing the UI Testing number column UI 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#023535",
      "width": "25%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="1"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="2"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        2
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="3"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        3
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing the UI Testing special button UI 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
    testID="1"
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;

exports[`Testing the UI Testing special functions UI 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#008F8C",
      "flex": 0.06,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.3,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="DEL"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        DEL
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.3,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
      testID="CE"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
          }
        }
      >
        CE
      </Text>
    </View>
  </View>
</View>
`;

exports[`Testing the UI Testing whole App UI 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8AA6A3",
        "flex": 0.1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
      testID="header"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#8AA6A3",
        "flex": 0.3,
        "justifyContent": "flex-end",
        "paddingBottom": 10,
        "paddingRight": 20,
        "paddingTop": null,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 50,
        }
      }
      testID="currNum"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#008F8C",
        "flex": 0.06,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0.3,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
        testID="DEL"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 30,
            }
          }
        >
          DEL
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.3,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
        testID="CE"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 30,
            }
          }
        >
          CE
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#127369",
        "flex": 0.6,
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#023535",
          "width": "25%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="7"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            7
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="4"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            4
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="1"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="."
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            .
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#023535",
          "width": "25%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="8"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            8
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="5"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            5
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="2"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            2
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="0"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            0
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#023535",
          "width": "25%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="9"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            9
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="6"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            6
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="3"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            3
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="="
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            =
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#008F8C",
          "width": "25%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="( - )"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            ( - )
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="+"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            +
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="-"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            -
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="x"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            x
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
          testID="/"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 30,
              }
            }
          >
            /
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
